docs/architecture/adr-024-pubsub.md

```diff
@@ -0,0 +1,80 @@
+# ADR 024: Pubsub
```

Comment:
Use https://github.com/tendermint/tendermint/issues/2313. I just added this one, it's 033

Reply:
Thank you 🍑